pkg/discovery/query.go

```diff
@@ -94,39 +87,51 @@ func QueryCluster(restConf *rest.Config, cfg *config.Config) error {
 		logrus.Errorf("Failed to query server data: %v", err)
 	}
 
+	// Get the list of namespaces and apply the regex filter on the namespace
+	logrus.Infof("Filtering namespaces based on the following regex:%s", cfg.Filters.Namespaces)
+	nslist, err := FilterNamespaces(kubeClient, cfg.Filters.Namespaces, cfg.Namespace)
```

Comment:
New param in this function allows for some fallback logic to consider that hardcoded namespace name.